src/index.ts

```diff
@@ -0,0 +1,94 @@
+import core from '@actions/core';
+import github from '@actions/github';
+import { Bumper } from 'conventional-recommended-bump';
+import { getLatestRelease, getNewVersion } from './utils.js';
+
+async function run() {
+  try {
+    core.startGroup('Finding last tag...');
+
+    // get the last tag
+    const octokit = github.getOctokit(core.getInput('repo-token'));
+
+    const repo = process.env.ACT
+      ? {
+          // for testing locally with act
+          owner: 'agrc',
+          repo: 'get-next-version-action',
+        }
+      : {
+          owner: github.context.payload.repository?.owner.login,
+          repo: github.context.payload.repository?.name,
+        };
+    core.info(`querying tags for ${JSON.stringify(repo)}`);
+
+    type GraphQLResponse = {
+      repository: {
+        releases: {
+          edges: {
+            node: {
+              id: string;
+              isPrerelease: boolean;
+              tag: {
+                name: string;
+              };
+            };
+          }[];
+        };
+      };
+    };
```

Comment:
**question**: i thought octokit had types. Why do we redefine this?

Reply:
It does have types, but I don't think that it would be possible for them to dynamically build a type based on the input query, would it? The type that they return is just [an object with any as a value](https://github.com/copperwall/graphql.js/blob/master/src/types.ts#L38-L40).